repository/src/main/java/org/alfresco/repo/preference/script/ScriptPreferenceService.java

```diff
@@ -76,8 +77,16 @@ public NativeObject getPreferences(String userName)
     public NativeObject getPreferences(String userName, String preferenceFilter)
     {
         // It's a tad unusual to return a NativeObject like this - at least within Alfresco.
-        // But we can't change it to e.g. a ScriptableHashMap as the API is published.
-        Map<String, Serializable> prefs = this.preferenceService.getPreferences(userName, preferenceFilter);        
+        // But we can't change it to e.g. a ScriptableHashMap as the API is published.
+        Map<String, Serializable> prefs = this.transactionService.getRetryingTransactionHelper().doInTransaction(new RetryingTransactionCallback<Map<String, Serializable>>()
+        {
+            @Override
+            public Map<String, Serializable> execute() throws Throwable
+            {
+                return preferenceService.getPreferences(userName, preferenceFilter);
+            }
+        }, true, true);
```

Comment:
Why do you need to create a new transaction here? It is usually a very dangerous approach. Especially, if it will start a new transaction within a thread that already has a transaction, this may result in a deadlock inside a DB.